Fix docker import test

duglin opened the pull request:
For me when I run the test I see:

```
Downloading from http://nourl/bad
Importing    283 B
Untar re-exec error: exit status 1: output: unexpected EOF
```

and nothing about "dial tcp" so it appears that the output is
system dependent and therefore we can't really check it. I think
checking for non-zero exit code is sufficient so I'm removing this
string check.

Signed-off-by: Doug Davis dug@us.ibm.com

Commit:
Fix docker import tests

For me when I run the test I see:
```
Downloading from http://nourl/bad
Importing    283 B
Untar re-exec error: exit status 1: output: unexpected EOF
```
and nothing about "dial tcp" so it appears that the output is
system dependent and therefore we can't really check it. I think
checking for non-zero exit code is sufficient so I'm removing this
string check.

Signed-off-by: Doug Davis <dug@us.ibm.com>

LK4D4 commented:
@duglin EOF looks really bad, I'm having this from time to time too. I think it's some real error.

duglin commented:
I'm assuming the EOF is due to the fact that, in my system, I actually get back:
```
HTTP/1.1 301 Moved Permanently
Server: nginx
Date: Mon, 23 May 2016 17:14:10 GMT
Content-Type: text/html
Transfer-Encoding: chunked
Connection: close
Location: http://localhost
Expires: Mon, 23 May 2016 17:14:09 GMT
Cache-Control: no-cache

<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
```
and not some tcp level error, and then this output isn't a valid tar file so we generate an EOF.
I guess we could look for a nicer error message.

LK4D4 commented:
Yeah, I think we should make error nicer. It's really weird.

duglin commented:
ok tweaked the error message to be a bit more user-friendly but that doesn't change the situation much since we can still get two different types of errors depending on the code path - the "dial" one and the new one I just added. So I modified the testcase to allow for either.

thaJeztah commented:
LGTM

icecrime commented:
LGTM
